### PR TITLE
feat:  Add release workflow and config

### PR DESCRIPTION
**Pull request by MartinWitt**
This is a jreleaser workflow we already use for maven lockfile. The secrets are set, see 
image
There workflow itself is pretty easy. You have a dropdown menu and decide if you want a patch/minor/major release. JReleaser updates the version and does the release to maven-central and github. If you want some extra feature have a look at https://jreleaser.org/guide/latest/reference/index.html. There are a lot of possibilities.

### CODE REVIEW

**Review by monperrus — APPROVED** [review]
thanks a lot

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,5 @@ gradle-app.setting
 
 ## Gradle Enterprise ID File ##
 .mvn/.gradle-enterprise/gradle-enterprise-workspace-id
+# Jreleaser files
+out/
```
Comment (I-Al-Istannen) [inline .gitignore:61]: newline :)

File: jreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+project:
+  name: spoon-core
+  description: A plugin to lock and validate the integrity of the dependencies of a maven based project.
```
Comment (I-Al-Istannen) [inline jreleaser.yml:3]: You sure?

File: jreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  longDescription: Spoon is an open-source library to analyze, rewrite, transform, transpile Java source code.
```
Comment (I-Al-Istannen) [inline jreleaser.yml:4]: Are you sure indentation in YAML works like this? I have no idea, but it looks a bit weird in github.

File: jreleaser.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  authors:
+    - slarse
+    - monperrus
+    - nharrand
+    - martinwitt
+    - sirywell
+    - I-Al-Istannen
+  license: (MIT OR CECILL-C)
+  links:
+    homepage: https://github.com/chains-project/maven-lockfile
```
Comment (I-Al-Istannen) [inline jreleaser.yml:16]: Not quite

File: jreleaser.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+deploy:
+  maven:
+    nexus2:
+      maven-central:
+        active: ALWAYS
+        url: https://s01.oss.sonatype.org/service/local
+        closeRepository: true
+        releaseRepository: true
+        stagingRepositories:
+          - target/staging-deploy
```
Comment (I-Al-Istannen) [inline jreleaser.yml:46]: newline

**Review by I-Al-Istannen — CHANGES_REQUESTED** [review]

File: jreleaser.yml (hunk 3 above)
Comment (MartinWitt) [reply jreleaser.yml:4]: The file with the same formatting works for maven lockfile. So, I'm confident it should work.

File: .github/workflows/jreleaser.yml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        run: echo "CURRENT_VERSION_WITH_SNAPSHOT=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)" >> $GITHUB_ENV
+# Calculate next version: 
+# - if version is patch, we just increment the patch version
+# - if version is minor or major, we increment the minor or major version and set the patch version to 0
+# As we are using a snapshot version, we need to run semver next twice to get the next version for mahor and minor releases. Reason: Any X.Y.Z-SNAPSHOT version will be released as X.Y.Z if we run semver next major X.Y.Z-SNAPSHOT
+      - name: Set next version
+        if: ${{ github.event.inputs.version == 'patch' }}
+        run: echo "NEXT_VERSION=$(semver next ${{ github.event.inputs.version }} $CURRENT_VERSION_WITH_SNAPSHOT)" >> $GITHUB_ENV
+      - name: Set next version
+        # semver next for a snapshot only releases the snapshot version, so we need to run it 2 times
```
[the remark this answers is not in the document]
Comment (MartinWitt) [reply .github/workflows/jreleaser.yml]: I have updated the doc. The action handles the version and updates after each release to next version + patch + snapshot. 

For example our current version is 10.4.1-SNAPSHOT.
For patch we release `10.4.1` and set the next version to `10.4.2-SNAPSHOT`
For minor we release `10.5.0` and set the next version to `10.5.1-SNAPSHOT`
For major we release `11.0.0` and set the next version to `11.0.1-SNAPSHOT`

Does this bring clarity?

File: jreleaser.yml (hunk 4 above)
Comment (MartinWitt) [reply jreleaser.yml]: Do we want the github page or [the spoon page](https://spoon.gforge.inria.fr/) here?

File: .github/workflows/jreleaser.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (I-Al-Istannen) [reply .github/workflows/jreleaser.yml]: Oh, what we want to do is clear to me. What is not clear to me is why we handle minor/major/patch differently. 
i.e. what `        # semver next for a snapshot only releases the snapshot version, so we need to run it 2 times` means.

I would expect us to do the following.
### For a patch
`semver next patch VERSION_WITH_SNAPSHOT`

### For a minor
`semver next minor` (this probably clears the patch) and then `semver next patch ++ "-SNAPSHOT"` (to inc the patch and append snapshot)

### For a major
`semver next major` (this probably clears the patch) and then `semver next patch ++ "-SNAPSHOT"` (to inc the patch and append snapshot)

---

My question is why you need to do one with the snapshot and one without, does `semver next minor A.B.C-SNAPSHOT` not strip off the snapshot?

[the remark this answers is not in the document]
Comment (MartinWitt) [reply .github/workflows/jreleaser.yml]: `> semver next patch 10.2.1-SNAPSHOT`
`10.2.1`
`> semver next major 10.2.1-SNAPSHOT`
`10.2.1`
If it is a snapshot, the plugin simple strips the snapshot and calls it day.

File: jreleaser.yml (hunk 4 above)
Comment (I-Al-Istannen) [reply jreleaser.yml]: I would say spoon page

**Review by I-Al-Istannen — APPROVED** [review]

**Comment by monperrus** [comment]
thanks a lot @MartinWitt , we'll try the workflow  together next week.

**Pull request merged**